modules/reference/pages/cluster-properties.adoc

```diff
@@ -1,1796 +1,5372 @@
-*Restart required*: yes
+*Visibility:* `tunable`
 
----
+*Type:* integer
 
-== Cluster management properties
+*Accepted values:* [`0`, `18446744073709551615`]
 
-=== cluster_id
+*Default:* `268435456`
 
-Cluster identifier.
+---
 
-*Type*: string
+=== compaction_ctrl_backlog_size
 
-*Default*: null
+Target backlog size for compaction controller. if not set compaction target compaction backlog would be equal to .
 
-*Restart required*: no
+*Requires Restart:* No
 
----
+*Nullable:* Yes
 
-=== controller_snapshot_max_age_sec
+*Visibility:* `tunable`
 
-Max time that will pass before Redpanda attempts to create a controller snapshot after a new controller command appears.
+*Type:* integer
 
-*Units*: seconds
+*Default:* `null`
 
-*Default*: 60
+---
 
-*Restart required*: no
+=== compaction_ctrl_d_coeff
 
----
+Derivative coefficient for compaction PID controller.
 
-=== enable_auto_rebalance_on_node_add
-{badge-deprecated}
+*Requires Restart:* No
 
-Enable automatic partition rebalancing when new nodes are added.
+*Nullable:* No
 
-*Type*: boolean
+*Visibility:* `tunable`
 
-*Default*: false
+*Type:* number
 
-*Restart required*: no
+*Default:* `0.2`
 
 ---
 
-=== enable_cluster_metadata_upload_loop
+=== compaction_ctrl_i_coeff
 
-Enables cluster metadata uploads. Required for xref:manage:whole-cluster-restore.adoc[whole cluster restore].
+Integral coefficient for compaction PID controller.
 
-*Type*: boolean
+*Requires Restart:* No
+
+*Nullable:* No
+
+*Visibility:* `tunable`
 
-*Default*: true
+*Type:* number
 
-*Restart required*: yes
+*Default:* `0.0`
 
 ---
 
-=== enable_controller_log_rate_limiting
+=== compaction_ctrl_max_shares
 
-Flag to enable limiting the write rate for the controller log.
+Maximum number of IO and CPU shares that compaction process can use.
 
-*Type*: boolean
+*Requires Restart:* No
 
-*Default*: false
+*Nullable:* No
 
-*Restart required*: no
+*Visibility:* `tunable`
 
----
+*Type:* integer
 
-=== enable_leader_balancer
+*Accepted values:* [`-32768`, `32767`]
 
-Enable automatic leadership rebalancing. Mode is set by <<leader_balancer_mode,`leader_balancer_mode`>>.
+*Default:* `1000`
 
-*Type*: boolean
+---
 
-*Default*: true
+=== compaction_ctrl_min_shares
 
-*Restart required*: no
+Minimum number of IO and CPU shares that compaction process can use.
 
----
+*Requires Restart:* No
 
-=== enable_rack_awareness
+*Nullable:* No
 
-Enable rack-aware replica assignment.
+*Visibility:* `tunable`
 
-*Type*: boolean
+*Type:* integer
 
-*Default*: false
+*Accepted values:* [`-32768`, `32767`]
 
-*Restart required*: no
+*Default:* `10`
 
 ---
 
-=== leader_balancer_mode
+=== compaction_ctrl_p_coeff
 
-Mode of the leader balancer for optimizing movements of leadership between shards (logical CPU cores). Enabled by <<enable_leader_balancer,`enable_leader_balancer`>>.
+Proportional coefficient for compaction PID controller. This has to be negative since compaction backlog should decrease when number of compaction shares increases.
```

Comment:
```suggestion
Proportional coefficient for compaction PID controller. This must be negative, because the compaction backlog should decrease when the number of compaction shares increases.
```